crates/wasmtime/src/runtime/gc/enabled/arrayref.rs

```diff
@@ -385,14 +498,121 @@ impl ArrayRef {
             allocator.store_id,
             "attempted to use a `ArrayRefPre` with the wrong store"
         );
+        assert!(
+            !store.async_support(),
+            "use `ArrayRef::new_fixed_async` with asynchronous stores"
+        );
```

Comment:
Could this get sunk into `retry_after_gc`? (and the dual into `retry_after_gc_async`?)

Reply:
Not without either making the panic message worse for users or plumbing through extra arguments to the retry functions that are just for the panic message. Neither seems worth the trade off.

Reply:
Personally I think it's fine to make the panic messages worse given the predicted rarity they'll come up, but I'm ok either way.